### PR TITLE
FOCUS #365: Updated the changelog with all changes

### LINKED ISSUE #365: Changelog appendix
Proposed template per release based on [keepachangelog.com](https://keepachangelog.com):

```markdown
## {version}

**➕ Added:**
- Foo column was added.

**✏️ Changed:**
- Bar attribute requirements were updated with the following:
  - Bar values MUST...
  - Bar values SHOULD...

**🚩 Deprecated:**
- Baz column is now deprecated and will be removed in version 2.0.

**🗑️ Removed:**
- Qux column has been removed.

**🛠️ Fixed:**
- Quux column description was updated to fix a grammatical error.
```

### Definition of done

- [ ] Create a "Changelog" appendix
- [ ] Create a PR template that confirms that people have met all process requirements, like updating the changelog

<br>

### Give it a :+1: below ↴

If you can discuss and help finalize the change, add yourself as an assignee :arrow_upper_right:

Comments are welcome and encouraged!

### PR DESCRIPTION
**Pull request by flanakin**
Resolves #365

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+## Unreleased
+
+**Added:**
+
+- `RegionId` column
+
+**Changed:**
+
+- `Column naming convention` attribute includes the following new provisions:
```
Comment (flanakin) [inline CHANGELOG.md:15]: Do we want to track new requirements in an existing spec as an addition or a change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  - Column IDs should not exceed 50 characters.
+- `EffectiveCost` column updated to clarify it does not mix or "blend" costs across multiple charges.
```
Comment (flanakin) [inline CHANGELOG.md:17]: Periods in some bullets and not in others is bothering me. How would others feel if the main bullet is the specific asset we're updating and the change description was always in a nested bullet? That is needed when there are multiple changes but we could apply it consistently for single changes as well. Thoughts?
```suggestion
- `EffectiveCost` column
  - Updated to clarify it does not mix or "blend" costs across multiple charges.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+- `Region` column renamed to `RegionName`.
+- `Tags` column updated to include:
+  - Tag keys that cannot have a value must use a boolean `true` as the tag value.
+
+**Fixed:**
```
Comment (flanakin) [inline CHANGELOG.md:22]: FYI - I intentionally did not include all text changes here. I only included things that changed the implementation details. If anyone wants the lower-level text changes, they can get it from the "all changes" link in each section.

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  - FOCUS columns must be listed before custom columns in the dataset.
+  - Added "SKU" as an exception to the "no acronyms" rule. The display name is "SKU" and column ID is `Sku`.
+- `AmortizedCost` column renamed to `EffectiveCost`
+- `ChargeType` column renamed to `ChargeCategory`
+
+[All 1.0-preview changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/v0.5-cr...v1.0-preview-cr)
+
+<br>
+
+## v0.5
```
Comment (flanakin) [inline CHANGELOG.md:72]: Originally, I wasn't planning to cover 0.5 but it felt strange to talk about GA relative to preview when preview didn't have a baseline to compare it to. This make the changelog complete for all announced versions.

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+**Added:**
+
+- `RegionId` column
+
+**Changed:**
+
+- `Column naming convention` attribute includes the following new provisions:
+  - Column IDs should not exceed 50 characters.
+- `EffectiveCost` column updated to clarify it does not mix or "blend" costs across multiple charges.
+- `Region` column renamed to `RegionName`.
```
Comment (flanakin) [inline CHANGELOG.md:18]: Should we track this as a change or an addition and removal? Ideally, it should align to the "Introduced version" in the columns.

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
+# FinOps Open Cost and Usage Specification Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
```
Comment (flanakin) [inline CHANGELOG.md:7]: Not sure why the preview is failing, but this is what it looks like when rendered...
---

**Added:**

- `RegionId` column

**Changed:**

- `Column naming convention` attribute includes the following new provisions:
  - Column IDs should not exceed 50 characters.
- `EffectiveCost` column updated to clarify it does not mix or "blend" costs across multiple charges.
- `Region` column renamed to `RegionName`.
- `Tags` column updated to include:
  - Tag keys that cannot have a value must use a boolean `true` as the tag value.

**Fixed:**

- `CommitmentDiscountType` column constraints were updated to show the column as required to align with the normative text.

[All unreleased changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/v1.0-preview-cr...working_draft)

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+**Fixed:**
+
+- `CommitmentDiscountType` column constraints were updated to show the column as required to align with the normative text.
+
+[All unreleased changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/v1.0-preview-cr...working_draft)
+
+<br>
+
+## v1.0-preview
```
Comment (flanakin) [inline CHANGELOG.md:30]: Not sure why the preview is failing, but this is what it looks like when rendered...

---

## v1.0-preview
<sup>Announced November 15, 2023</sup>

**Added:**

- `Discount handling` attribute
- `Key/value format` attribute
- `Numeric format` attribute
- `Unit format` attribute
- `ChargeDescription` column
- `ChargeFrequency` column
- `ChargeSubcategory` column
- `CommitmentDiscountCategory` column
- `CommitmentDiscountId` column
- `CommitmentDiscountName` column
- `CommitmentDiscountType` column
- `ListCost` column
- `ListUnitPrice` column
- `PricingCategory` column
- `PricingQuantity` column
- `PricingUnit` column
- `ResourceType` column
- `SkuId` column
- `SkuPriceId` column
- `Tags` column
- `UsageQuantity` column
- `UsageUnit` column

**Changed:**

- `Column naming convention` attribute includes the following new provisions:
  - Columns with a `Category` suffix must be normalized.
  - External, custom columns can be added but must be prefixed by `x_`.
  - FOCUS columns must be listed before custom columns in the dataset.
  - Added "SKU" as an exception to the "no acronyms" rule. The display name is "SKU" and column ID is `Sku`.
- `AmortizedCost` column renamed to `EffectiveCost`
- `ChargeType` column renamed to `ChargeCategory`

[All 1.0-preview changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/v0.5-cr...v1.0-preview-cr)

File: CHANGELOG.md (hunk 4 above)
Comment (flanakin) [inline CHANGELOG.md:72]: Not sure why the preview is failing, but this is what it looks like when rendered...

---

## v0.5
<sup>Announced June 24, 2023</sup>

**Added:**
- `Column naming convention` attribute
- `Currency code format` attribute
- `Date/time format` attribute
- `Null handling` attribute
- `AmortizedCost` column
- `AvailabilityZone` column
- `BilledCost` column
- `BillingAccountId` column
- `BillingAccountName` column
- `BillingCurrency` column
- `BillingPeriodEnd` column
- `BillingPeriodStart` column
- `ChargePeriodEnd` column
- `ChargePeriodStart` column
- `ChargeType` column
- `InvoiceIssuerName` column
- `ProviderName` column
- `PublisherName` column
- `Region` column
- `ResourceId` column
- `ResourceName` column
- `ServiceCategory` column
- `ServiceName` column
- `SubAccountId` column
- `SubAccountName` column

[All 0.5 changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/7106bbe...v0.5-cr)

**Review by ahullah — APPROVED** [review]

**Review by mike-finopsorg — APPROVED** [review]

**Comment by mike-finopsorg** [comment]
Approved at weekly meeting 11/04/2023

**Comment by udam-f2** [comment]
4/11: The group decided that this is ready for merge. There are incremental improvements that can be made to the change descriptions (and possibly a set of principles that we introduce - e.g. if normative changes happened, must be listed). However, since it's not going into the specification, those changes can happen prior to the release of v1.0.

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+## Unreleased
+
+**Added:**
+
+- `RegionId` column
+
+**Changed:**
+
+- `Column naming convention` attribute includes the following new provisions:
+  - Column IDs should not exceed 50 characters.
```
Comment (flanakin) [inline CHANGELOG.md:16]:
```suggestion
  - Column IDs should not exceed 50 characters.
- `CommitmentDiscountCategory` column updates:
  - Column is conditional and only required when the provider supports commitment discounts.
- `CommitmentDiscountId` column updates:
  - Column is conditional and only required when the provider supports commitment discounts.
- `CommitmentDiscountName` column updates:
  - Column is conditional and only required when the provider supports commitment discounts.
- `CommitmentDiscountType` column updates:
  - Column is conditional and only required when the provider supports commitment discounts.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+- `EffectiveCost` column updated to clarify it does not mix or "blend" costs across multiple charges.
+- `Region` column renamed to `RegionName`.
+- `Tags` column updated to include:
+  - Tag keys that cannot have a value must use a boolean `true` as the tag value.
+
+**Fixed:**
+
+- `CommitmentDiscountType` column constraints were updated to show the column as required to align with the normative text.
+
```
Comment (flanakin) [inline CHANGELOG.md:25]:
```suggestion

**Removed:**

- `ChargeSubcategory` column

```

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+
+**Added:**
+
+- `RegionId` column
```
Comment (flanakin) [inline CHANGELOG.md:11]:
```suggestion
- `ContractedCost` column
- `ContractedUnitPrice` column
- `RegionId` column
```

File: CHANGELOG.md (hunk 5 above)
Comment (flanakin) [reply CHANGELOG.md:18]: Changing to treat this as a removal
```suggestion
```

**Comment by jpradocueva** [comment]
Approved by the Group on April 18.

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+- `ServiceName` column
+- `SubAccountId` column
+- `SubAccountName` column
+
+[All 0.5 changes](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/compare/7106bbe...v0.5-cr)
+
+## Columns History
+This table maps the evolution of the billing data, showcasing column introductions and updates from its initial version 0.5 to the comprehensive revisions in "1.0-preview", "1.0" and beyond.
+
+| **Column Title**                  | **Revision Introduced** | **Latest Update**
```
Comment (flanakin) [inline CHANGELOG.md:155]:
```suggestion
| **Column Name**                   | **Revision Introduced** | **Latest Update**
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+|-----------------------------------|-------------------|--------------------------|
+| Availability Zone                             | 0.5                          |                                          |                  
+| Billed Cost                      | 0.5               |                                            |
```
Comment (flanakin) [inline CHANGELOG.md:158]:
```suggestion
| Availability Zone                 | 0.5                          |                                          |                  
| Billed Cost                      | 0.5               |                                            |         
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+
+**Added:**
+
+-  `ChargeClass` column
```
Comment (flanakin) [inline CHANGELOG.md:11]:
```suggestion
- `ChargeClass` column
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+| Resource Type                    | 1.0-preview              |                                          |    
+| Service Category                 | 0.5               |                                          |    
+| Service Name                     | 0.5               |                                          |    
+| SKU ID                           | 1.0-preview             |                                          |    
+| SKU Price ID                     | 1.0-preview              |                                          |    
+| Sub Account ID                   | 0.5               |                                          |    
+| Sub Account Name                 | 0.5               |                                          |    
+| Tags                             | 1.0-preview               |                                          |    
+| Usage Quantity                   | 1.0-preview             |                                          |    
+| Usage Unit                       | 1.0-preview              |                                          |
```
Comment (flanakin) [inline CHANGELOG.md:198]:
```suggestion
| **Column ID**                | **Revision Introduced** | **Latest Update** |
|------------------------------|-------------------------|-------------------|
| AvailabilityZone             | 0.5                     |                   |
| BilledCost                   | 0.5                     |                   |
| BillingAccountId             | 0.5                     |                   |
| BillingAccountName           | 0.5                     |                   |
| BillingCurrency              | 0.5                     |                   |
| BillingPeriodEnd             | 0.5                     |                   |
| BillingPeriodStart           | 0.5                     |                   |
| ChargeCategory               | 0.5                     | 1.0               |
| ChargeClass                  | 1.0                     |                   |
| ChargeDescription            | 1.0-preview             |                   |
| ChargeFrequency              | 1.0-preview             |                   |
| ChargePeriodEnd              | 0.5                     |                   |
| ChargePeriodStart            | 0.5                     |                   |
| ChargeSubcategory            | 1.0-preview             | Removed in 1.0    |    
| CommitmentDiscountCategory   | 1.0-preview             |                   |
| CommitmentDiscountId         | 1.0-preview             |                   |
| CommitmentDiscountName       | 1.0-preview             |                   |
| CommitmentDiscountType       | 1.0-preview             |                   |
| ContractedCost               | 1.0                     |                   |
| ContractedUnitPrice          | 1.0                     |                   |
| EffectiveCost                | 0.5                     |                   |    
| InvoiceIssuerName            | 0.5                     |                   |    
| ListCost                     | 1.0-preview             |                   |
| ListUnitPrice                | 1.0-preview             |                   |
| PricingCategory              | 1.0-preview             |                   |
| PricingQuantity              | 1.0-preview             |                   |
| PricingUnit                  | 1.0-preview             |                   |
| ProviderName                 | 0.5                     |                   |
| PublisherName                | 0.5                     |                   |
| Region                       | 0.5                     | Removed in 1.0    |
| RegionId                     | 1.0                     |                   |
| RegionName                   | 1.0                     |                   |
| ResourceId                   | 0.5                     |                   |
| ResourceName                 | 0.5                     |                   |
| ResourceType                 | 1.0-preview             |                   |
| ServiceCategory              | 0.5                     |                   |
| ServiceName                  | 0.5                     |                   |
| SkuId                        | 1.0-preview             |                   |
| SkuPriceId                   | 1.0-preview             | 1.0               |
| SubAccountId                 | 0.5                     |                   |
| SubAccountName               | 0.5                     |                   |
| Tags                         | 1.0-preview             | 1.0               |
| UsageQuantity                | 1.0-preview             |                   |
| UsageUnit                    | 1.0-preview             |                   |
```

**Pull request merged**